src/ctapipe/calib/camera/calibrator.py

```diff
@@ -292,13 +296,21 @@ def _calibrate_dl1(self, event, tel_id):
             and dl1_calib.absolute_factor is not None
         ):
             if selected_gain_channel is None:
```

Comment:
I think this section could use some refactoring to make it easier to follow.

I think the logic is the same if we simplify to:

```python
if selected_gain_channel is None:
    calibration = dl1_calib.relative_factor / dl1_calib.absolute_factor
else:
        calibratoin = (
            dl1_calib.relative_factor[selected_gain_channel, pixel_index]
             / dl1_calib.absolute_factor[selected_gain_channel, pixel_index]
        )

if isinstance(extractor, VarianceExtractor):
    corr = corr**2

dl1.image *= corr
```

Reply:
Sounds good, I'll get to it when I arrive home

Reply:
i put it in